src/components/counter/counter.test.tsx

```diff
@@ -0,0 +1,29 @@
+import { describe, expect, test } from 'vitest'
+import { render } from '@testing-library/react'
+import '@testing-library/jest-dom'
+import { Counter } from './counter'
+import userEvent from '@testing-library/user-event'
+
+const user = userEvent.setup()
+
+describe('Counter Component', () => {
+  test('Test that component renders initial value provided in props', () => {
+    const { getByText } = render(<Counter initialValue={7} />)
```

Comment:
Creators of RTL recommend using `screen`. Like in this example - https://testing-library.com/docs/queries/about/#example

https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-screen

```suggestion
    render(<Counter initialValue={7} />)
```